frontend/test/pytest/test_transform.py

```diff
@@ -0,0 +1,473 @@
+    data = data[0]
+    data = pnp.array([data, data, data, data, data])
+    x = pnp.array([0.1, 0.1, 0.1, 0.1, 0.1], requires_grad=True)
+    weights = pnp.ones((batch_size, 10, 3, 3), requires_grad=True)
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+    expected = jax_jit(data, x, weights)
+    observed = compiled(data, x, weights)
+    assert np.allclose(expected, observed)
+
+
+def test_split_non_commuting(backend):
+    """Test split non commuting"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.split_non_commuting
+        @qml.qnode(qml.device(device_name, wires=6), interface="jax")
+        def qfunc():
+            """Example taken from PL tests"""
+            qml.Hadamard(1)
+            qml.Hadamard(0)
+            qml.PauliZ(0)
+            qml.Hadamard(3)
+            qml.Hadamard(5)
+            qml.T(5)
+            return (
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliX(0)),
+                qml.expval(qml.PauliZ(1)),
+                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+                qml.expval(qml.PauliX(3)),
+                qml.expval(qml.PauliY(5)),
+            )
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+    expected = jax.jit(qnode_control)()
+    observed = qjit(qnode_backend)()
+    assert np.allclose(expected, observed)
+
+
+class RX_broadcasted(qml.RX):
+    """A version of qml.RX that detects batching."""
+
+    ndim_params = (0,)
+    compute_decomposition = staticmethod(lambda theta, wires=None: [qml.RX(theta, wires=wires)])
+
+
+class RZ_broadcasted(qml.RZ):
+    """A version of qml.RZ that detects batching."""
+
+    ndim_params = (0,)
+    compute_decomposition = staticmethod(lambda theta, wires=None: [qml.RZ(theta, wires=wires)])
+
+
+parameters = [
+    (0.2, np.array([0.1, 0.8, 2.1]), -1.5),
+    (0.2, np.array([0.1]), np.array([-0.3])),
+    (
+        0.2,
+        pnp.array([0.1, 0.3], requires_grad=True),
+        pnp.array([-0.3, 2.1], requires_grad=False),
+    ),
+]
+
+coeffs0 = [0.3, -5.1]
+H0 = qml.Hamiltonian(qml.math.array(coeffs0), [qml.PauliZ(0), qml.PauliY(1)])
+
+observables = [
+    [qml.PauliZ(0)],
+    [qml.PauliZ(0) @ qml.PauliY(1)],
+    [qml.PauliZ(0), qml.PauliY(1)],
+    [H0],
+]
+
+
+class TestBroadcastExpand:
+    """Test Broadcast Expand"""
+
+    @pytest.mark.skip(reason="https://github.com/PennyLaneAI/pennylane/issues/2762")
+    @pytest.mark.parametrize("params", parameters)
+    @pytest.mark.parametrize("obs", observables)
+    def test_expansion_qnode(self, backend, params, obs):
+        """Test broadcast expand"""
+
+        if obs[0] == H0:
+            pytest.skip(reason="https://github.com/PennyLaneAI/catalyst/issues/339")
```

Comment:
I'm wondering if we could generate an xfail here, because that way it will be easier to realize when the issue is fixed.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/280/commits/a438eba2e8a5db066557f115dec7de7c2bd37c55